BraveRewards/Ledger/BATBraveLedger.mm

```diff
@@ -757,7 +834,205 @@ - (void)confirmationsTransactionHistoryDidChange
 
 #pragma mark - Notifications
 
-// TODO: Add notification logic here
+- (NSArray<BATRewardsNotification *> *)notifications
+{
+  return [self.mNotifications copy];
+}
+
+- (void)clearNotificationWithID:(NSString *)notificationID
+{
+  for (BATRewardsNotification *n in self.notifications) {
+    if ([n.id isEqualToString:notificationID]) {
+      [self clearNotification:n];
+      return;
+    }
+  }
+}
+
+- (void)clearNotification:(BATRewardsNotification *)notification
+{
+  [self.mNotifications removeObject:notification];
+}
+
+- (void)clearAllNotifications
+{
+  [self.mNotifications removeAllObjects];
+}
+
+- (void)startNotificationTimers
+{
+  if (!self.isEnabled) {
+    return;
+  }
+  
+  // Startup timer, begins after 30-second delay.
+  self.notificationStartupTimer =
+  [NSTimer scheduledTimerWithTimeInterval:30
+                                   target:self
+                                 selector:@selector(notificationTimerFired)
+                                 userInfo:nil
+                                  repeats:NO];
+  
+  // Periodic timer, runs once per day by default.
+  self.notificationCheckTimer =
+  [NSTimer scheduledTimerWithTimeInterval:60 * 60 * 24
```

Comment:
24 hours is a lot and the app may get killed by that time, I guess this is just a fallback timer.
And usually when user will open the app on next day `notificationStartupTimer` will launch after 30 seconds, right?

Reply:
Hmm, the logic was copied from brave-core's implementation but yeah perhaps we should change ours based on background/foregrounding?

But yeah, if the app is ejected from memory and relaunches it'll check again in 30s